react/react1/week3/react1week3-todos/src/components/TodoList.js

```diff
@@ -0,0 +1,73 @@
+// Todolist
+
+import { DataFetch } from './DataFetch';
+import { useState } from 'react';
+import { TodoItem } from './TodoItem';
+
+export const TodoList = () => {
+  const [fetchTodo, setFetchTodo] = useState([]);
+  const [description, setDescription] = useState('');
+  const [deadline, setDeadline] = useState('');
+
+  //add Todo
+  const addTodo = () => {
+    let newID;
+    if (fetchTodo.length === 0) {
+      newID = 1;
+    } else {
+      newID = fetchTodo[fetchTodo.length - 1].id + 1;
+    }
+    return setFetchTodo([...fetchTodo, { id: newID, description, deadline }]);
+  };
+
+  //todo List
+  const todoList = () => {
+    return fetchTodo.map((inputTodo) => {
+      return (
+        <TodoItem
+          description={inputTodo.description}
+          deadline={inputTodo.deadline}
+          key={inputTodo.id}
+          id={inputTodo.id}
+          deletetodo={deleteTodo}
+          updateTodo={updateTodo}
+        />
+      );
+    });
+  };
+
+  //delete a Todo
+  const deleteTodo = (id) => {
+    setFetchTodo((prevTodos) => {
+      const listAfterDelete = prevTodos.filter((todo) => todo.id !== id);
+      return listAfterDelete;
+    });
+  };
+
+  // update todo
+  const updateTodo = (id, description) => {
+    console.log(id, description);
+    const listAfterUpdate = [...fetchTodo].map((todo) => {
+      if (todo.id === id) {
+        todo.description = description;
+      }
+      return todo;
+    });
+    setFetchTodo(listAfterUpdate);
+  };
+
+  return (
+    <div>
+      <h1> All about Todos </h1>
+      <DataFetch setFetchTodo={setFetchTodo} />
```

Comment:
DataFetch is not a component because it is not returning anything. You should remove it from here. You are only using useEffect and function from it. Sou you have to call this DataFetch functions out of return. (Also name should be noncapital). So you can use dataFetch(setFetchTodo) function outside of return. It will fetch all todos. Please reach me on slack if you have any question.

Reply:
i did this can you please have a look one time. if that is ok